web/modules/custom/joinup_core/joinup_core.module

```diff
@@ -869,41 +869,23 @@ function joinup_core_rdf_entity_update(EntityInterface $entity) {
  * published, the creation date is updated with the current datetime.
  */
 function joinup_core_node_presave(EntityInterface $entity) {
-  // Documents do not show the created date in the UI as they have the document
-  // publication date to show instead. However, since the listing screens sort
-  // by created date, the documents are updated just as the rest of the
-  // community content.
-  $community_bundles = ['discussion', 'document', 'event', 'news'];
-  if ($entity->isNew() || !in_array($entity->bundle(), $community_bundles) || !$entity->isPublished()) {
-    return;
+  // By default, publication_date module will set the value of `published_at` to
+  // `PUBLICATION_DATE_DEFAULT` value when the entity is not published and there
+  // is no value set for the field. According to publication_date, this is so
+  // that these entities come first in the results. Set this to NULL in this
+  // case.
+  // @see \Drupal\publication_date\Plugin\Field\FieldType\PublicationDateItem::preSave
+  if ($entity->published_at->value === PUBLICATION_DATE_DEFAULT) {
+    $entity->set('published_at', NULL);
   }
-  // Set the update time as the created time only if this is the first time this
-  // entity is published.
-  $published_revisions = \Drupal::entityQuery('node')
-    ->allRevisions()
-    ->condition('type', $entity->bundle())
-    ->condition('status', 1)
-    ->condition('nid', $entity->id())
-    ->sort('vid', 'ASC')
-    ->range(0, 1)
-    ->execute();
 
-  if (count($published_revisions) === 0) {
-    $entity->set('created', \Drupal::time()->getRequestTime());
-  }
-  else {
-    // If the node is reverted into a previously created draft version, the
-    // created time is a date different than the first publication date. For
-    // that case, always set the created time from the first published revision
-    // every time a community content is saved.
-    $revision_ids = array_keys($published_revisions);
-    $revision_id = reset($revision_ids);
-    $revision = \Drupal::entityTypeManager()->getStorage('node')->loadRevision($revision_id);
-    if (!empty($revision)) {
-      $entity->set('created', $revision->getCreatedTime());
-    }
+  if (!$entity->isPublished()) {
+    return;
   }
 
+  if (empty($entity->published_at->value)) {
+    $entity->set('published_at', \Drupal::time()->getRequestTime());
+  }
```

Comment:
That is strictly needed because of my 
```
  if ($entity->published_at->value === PUBLICATION_DATE_DEFAULT) {
    $entity->set('published_at', NULL);
  }
```
a bit above.

The publication_date module is doing exactly the same but checking the value being equal to the default.
```
    elseif ($this->isPublished() && $this->value == PUBLICATION_DATE_DEFAULT) {
      $this->value = REQUEST_TIME;
    }
```
If we do not set it to NULL, we do not need this part either.